spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/web/mappings/reactive/HandlerFunctionDescription.java

```diff
@@ -29,7 +29,7 @@
 	private final String className;
 
 	HandlerFunctionDescription(HandlerFunction<?> handlerFunction) {
-		this.className = handlerFunction.getClass().getCanonicalName();
+		this.className = handlerFunction.getClass().getName();
```

Comment:
I don't think we should do this. Perhaps as a fallback if `getCanonicalName` is empty? It's a change of format so it's a breaking change. 

Or perhaps we could adapt the test? I haven't looked in details yet so thinking out loud.

Reply:
I wrote to the core-libs mailing list a week ago, I think, and got this reply.

> A hidden class has no canonical name [1] and it can't be nominally referenced [...] For diagnosiability, Class::getName or Class::toString can be used.

As `HandlerFunction` is often a lambda, it will probably often return null on JDK 15 at least. I  don't have a problem with the fallback idea. In reality, there will be only a change for inner classes though.  See the following example output from JDK 14 of canonical vs name:
```
com.example.demo.Example$$Lambda$14/0x0000000800b66c40
com.example.demo.Example$$Lambda$14/0x0000000800b66c40
com.example.demo.AFunction
com.example.demo.AFunction
com.example.demo.Example.InnerFunction
com.example.demo.Example$InnerFunction
```

Give me a minute to implement the fallback.

Reply:
Done